### ISSUE TITLE
Make no_std compatible

### ISSUE THREAD

**Issue by t-moe** [bengreenier/partially#7]
Thanks for this awesome crate.

In order to use it on no_std targets I had to modify the header of lib.rs

```
#![no_std]
#![doc = core::include_str!(core::concat!("../", core::env!("CARGO_PKG_README")))]
```

**Comment by bengreenier** [comment]
Thanks @t-moe - I don't have any experience with `no_std`, but is that the only change you had to make? If so, sounds easy enough to include here - do you know if we also need some `no_std` feature flag to enable/disable such an attribute?

**Comment by bengreenier** [comment]
Looking at [serde](https://github.com/search?q=repo%3Aserde-rs%2Fserde%20%23!%5Bno_std%5D&type=code) I don't think we should need `#![no_std]` in this library crate, only in the consuming application crate (in the serde example, they have it in the test)...

That said, I put together a minimal repro that builds a no_std binary for windows (which is easiest for me to test, at the moment) and do indeed see that partially isn't compatible (causes build error when used): https://github.com/bengreenier/test_partially_no_std/blob/main/src/main.rs

I'll have to play with this a bit more tomorrow though, as the phrasing at https://docs.rust-embedded.org/embedonomicon/smallest-no-std.html#what-does-no_std-mean coupled with the serde example suggests to me that we probably shouldn't just be dropping in `#![no_std]` as the fix here.

**Comment by bengreenier** [comment]
created a built-in test application to mirror serde's setup, so we now have a (failing to compile) test case for `no_std` in repo. https://github.com/bengreenier/partially/tree/feat/no_std/crates/test_no_std

**Comment by t-moe** [comment]
Serde has more to say about no_std: Look here: https://github.com/search?q=repo%3Aserde-rs%2Fserde+no_std&type=code

Here's some implementation suggestion: https://github.com/KodrAus/rust-no-std/blob/main/src/lib.rs
(via https://stackoverflow.com/questions/57611219/when-adding-no-std-to-a-library-are-there-any-disadvantages-or-complicati )

**Comment by bengreenier** [comment]
Just linking in some other resources I've come across while investigating this one:
- https://gitlab.com/jD91mZM2/no-std-compat
- https://www.lurklurk.org/effective-rust/no-std.html

Starting to sound like maybe it _is_ normal to just add `#![no_std]` to a library crate itself, and then flag any features that depend on `std` with an `std` feature that's enabled by default.

**Comment by bengreenier** [comment]
landed in `0.2.0`: [crates.io](https://crates.io/crates/partially/0.2.0) - [docs.rs](https://docs.rs/partially/0.2.0/partially/index.html)
